green_box/api/notifications.py

```
@@ -21,24 +21,40 @@ def post(body):
     logger.info("Notification received")
     logger.info(body)
 
-    # Get bundle uuid and version
-    uuid, version = extract_uuid_version(body)
+    # Get bundle uuid, version and subscription_id
+    uuid, version, subscription_id = extract_uuid_version_subscription_id(body)
+
+    # launch wdl based on subscription id
+    wdl = None
+    for each in green_config.wdls:
+        if each.get('subscription_id', None) == subscription_id:
+            wdl = each
+    if not wdl:
+        return response_with_server_header(dict(error='Bad Request'), 400)
 
     # Prepare inputs
-    inputs = compose_inputs(green_config.workflow_name, uuid, version, green_config.provenance_script)
+    workflow_name = wdl.get('workflow_name')
+    inputs = compose_inputs(workflow_name, uuid, version, green_config.provenance_script)
     with open('cromwell_inputs.json', 'w') as f:
         json.dump(inputs, f)
 
     # Start workflow
-    logger.info('Launching {0} workflow in Cromwell'.format(green_config.workflow_name))
-    result = subprocess.check_output(['gsutil', 'cp', green_config.wdl_cloud_path, '.'])
-    result = subprocess.check_output(['gsutil', 'cp', green_config.wdl_deps_cloud_path, '.'])
-    result = subprocess.check_output(['gsutil', 'cp', green_config.default_inputs_cloud_path, '.'])
-    cromwell_response = start_workflow(green_config.wdl_file, green_config.wdl_deps_file, 'cromwell_inputs.json', green_config.default_inputs_file, green_config)
+    logger.info(wdl)
+    logger.info('Launching {0} workflow in Cromwell'.format(workflow_name))
+
+    # Get files from gsc
+    subprocess.check_output(['gsutil', 'cp', wdl['wdl_cloud_path'], '.'])
+    subprocess.check_output(['gsutil', 'cp', wdl['default_inputs_cloud_path'], '.'])
+    subprocess.check_output(['gsutil', 'cp', wdl['wdl_deps_cloud_path'], '.'])
+
+    wdl_file = wdl.get('wdl_file', None)
```

It might be better to throw an error at this point if wdl_file or any of the other parameters below are missing. So you could use the wdl_file = wdl['wdl_file'] syntax to achieve that, or else check that all params are populated before calling cromwell below. If you keep the .get calls, I think it's best to not pass the default None argument.